services/consensuscontext/test/validate_block_test.go

```diff
@@ -10,154 +10,120 @@ import (
-	block.ResultsBlock.Header.MutateStateDiffHash(stateDiffHash)
-	block.ResultsBlock.Header.MutatePreExecutionStateMerkleRootHash(mockedPreExecutionStateMerkleRoot)
+func rxInputs(cfg config.ConsensusContextConfig) *services.ValidateResultsBlockInput {
+	block := builders.BlockPairBuilder().WithCfg(cfg).Build()
 
 	input := &services.ValidateResultsBlockInput{
-		CurrentBlockHeight: currentBlockHeight,
+		CurrentBlockHeight: block.TransactionsBlock.Header.BlockHeight(),
 		TransactionsBlock:  block.TransactionsBlock,
+		PrevBlockHash:      block.TransactionsBlock.Header.PrevBlockHashPtr(),
+		PrevBlockTimestamp: block.TransactionsBlock.Header.Timestamp() - 1000,
 		ResultsBlock:       block.ResultsBlock,
-		PrevBlockHash:      validPrevBlockHash,
-		PrevBlockTimestamp: validPrevBlockTimestamp,
 	}
 
 	return input
 }
 
 func TestValidateResultsBlockOnValidBlock(t *testing.T) {
-	log := log.DefaultTestingLogger(t)
-	metricFactory := metric.NewRegistry()
-	cfg := config.ForConsensusContextTests(nil)
-	txPool := &services.MockTransactionPool{}
-	txPool.When("ValidateTransactionsForOrdering", mock.Any, mock.Any).Return(nil, nil)
-
-	input := rxInputs(cfg)
-
-	vm := &services.MockVirtualMachine{}
-	output := &services.ProcessTransactionSetOutput{
-		TransactionReceipts: input.ResultsBlock.TransactionReceipts,
-		ContractStateDiffs:  input.ResultsBlock.ContractStateDiffs,
-	}
-
-	vm.When("ProcessTransactionSet", mock.Any, mock.Any).Return(output, nil)
-
-	stateStorage := &services.MockStateStorage{}
-	stateHashOutput := &services.GetStateHashOutput{
-		StateMerkleRootHash: input.ResultsBlock.Header.PreExecutionStateMerkleRootHash(),
-	}
-	stateStorage.When("GetStateHash", mock.Any, mock.Any).Return(stateHashOutput, nil)
-
-	s := consensuscontext.NewConsensusContext(
-		txPool,
-		vm,
-		stateStorage,
-		cfg,
-		log,
-		metricFactory)
-
-	_, err := s.ValidateResultsBlock(context.Background(), input)
-	require.NoError(t, err, "validation should succeed on valid block")
+	test.WithContext(func(ctx context.Context) {
+		log := log.DefaultTestingLogger(t)
+		metricFactory := metric.NewRegistry()
+		cfg := config.ForConsensusContextTests(nil, false)
+		txPool := &services.MockTransactionPool{}
+		txPool.When("ValidateTransactionsForOrdering", mock.Any, mock.Any).Return(nil, nil)
+
+		input := rxInputs(cfg)
+
+		vm := &services.MockVirtualMachine{}
+		output := &services.ProcessTransactionSetOutput{
+			TransactionReceipts: input.ResultsBlock.TransactionReceipts,
+			ContractStateDiffs:  input.ResultsBlock.ContractStateDiffs,
+		}
+
+		vm.When("ProcessTransactionSet", mock.Any, mock.Any).Return(output, nil)
+
+		stateStorage := &services.MockStateStorage{}
+		stateHashOutput := &services.GetStateHashOutput{
+			StateMerkleRootHash: input.ResultsBlock.Header.PreExecutionStateMerkleRootHash(),
+		}
+		stateStorage.When("GetStateHash", mock.Any, mock.Any).Return(stateHashOutput, nil)
+
+		s := consensuscontext.NewConsensusContext(
```

Comment:
Why no harness here?

Reply:
only copied code so didn't really look inside. however changed to the harness